File: Frontend/ExamSupportToolUI/src/app/shared/models/student/examination-session-for-student.ts
import { PresentationScheduleForStudent } from "./presentation-schedule-for-student";
import { StudentPresentationForStudent } from "./student-presentation-for-student";

export interface ExaminationSessionForStudent {
    id: string;
    name: string;
    startDate: string;
    endDate: string;
    status: number;
    studentPresentation?: StudentPresentationForStudent;
    presentationSchedule?: PresentationScheduleForStudent;
}

File: Backend/ExamSupportToolAPI/ExamSupportToolAPI.DataObjects/ExaminationSessionForStudent.cs
namespace ExamSupportToolAPI.DataObjects
{
    public class ExaminationSessionForStudent
    {
        public string Name { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int Status { get; set; }
        public StudentPresentationForStudent? StudentPresentation { get; set; }
        public PresentationScheduleForStudent? PresentationSchedule { get; set; }
        
    }
}

